frontend/src/pages/article/components/article-form/article-form.tsx

```diff
@@ -1,26 +1,48 @@
+import { Button, Input, Textarea } from '~/libs/components/components.js';
 import { ButtonType } from '~/libs/enums/enums.js';
-import { useAppDispatch, useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
 import { type ValueOf } from '~/libs/types/types.js';
 import {
   articleCreateValidationSchema,
   type ArticleRequestDto,
 } from '~/packages/articles/articles.js';
+import { PromptType } from '~/packages/prompts/prompts.js';
 import { actions as articlesActions } from '~/slices/articles/articles.js';
+import { actions as promptsActions } from '~/slices/prompts/prompts.js';
 
-import { Button, Input, Textarea } from '../components.js';
-import { DEFAULT_ARTICLE_FORM_PAYLOAD } from './libs/constants.js';
+import { DEFAULT_ARTICLE_FORM_PAYLOAD } from './libs/constants/constants.js';
 import { ArticleSubmitType } from './libs/enums/enums.js';
 import styles from './styles.module.scss';
 
-// TODO: Should be moved out from common components.
 const ArticleForm: React.FC = () => {
+  const { createdPrompt, generatedPrompt } = useAppSelector(({ prompts }) => ({
+    generatedPrompt: prompts.generatedPrompt,
+    createdPrompt: prompts.createdPrompt,
+  }));
   const dispatch = useAppDispatch();
   const { control, errors, handleSubmit, handleReset, isDirty, isSubmitting } =
     useAppForm<ArticleRequestDto>({
       defaultValues: DEFAULT_ARTICLE_FORM_PAYLOAD,
       validationSchema: articleCreateValidationSchema,
     });
 
+  useEffect(() => {
+    if (generatedPrompt) {
+      void dispatch(
+        promptsActions.createPrompt({
+          type: PromptType.MANUAL,
+          ...generatedPrompt,
+        }),
+      );
+    }
+  }, [dispatch, generatedPrompt]);
```

Comment:
useEffect is used here to save the prompt to the database if there is a generated prompt. But I'm not sure if it's correct. I tried to do that in handleArticleSubmit, but createdPrompt will be null after the dispatch createPrompt, I also tried to use async/await, but it doesn't help. Please suggest something here